Sources/AssociatedObjectPlugin/AssociatedObjectMacro.swift

```diff
@@ -131,18 +132,40 @@ extension AssociatedObjectMacro {
     static func getter(
         identifier: TokenSyntax,
         type: TypeSyntax,
+        policy: ExprSyntax,
         defaultValue: ExprSyntax?
     ) -> AccessorDeclSyntax {
         AccessorDeclSyntax(
             accessorSpecifier: .keyword(.get),
             body: CodeBlockSyntax {
-                """
-                objc_getAssociatedObject(
-                    self,
-                    &Self.__associated_\(identifier.trimmed)Key
-                ) as? \(type)
-                ?? \(defaultValue ?? "nil")
-                """
+                if let defaultValue {
+                    """
+                    if let value = objc_getAssociatedObject(
+                        self,
+                        &Self.__associated_\(identifier.trimmed)Key
+                    ) as? \(type.trimmed) {
+                        return value
+                    }
+                        let value = \(defaultValue.trimmed)
```

Comment:
It may be necessary to specify the type here as well

let value: \(type.trimmed) = \(defaultValue.trimmed)